core/services/log_zipper/main.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import glob
+import os
+import time
+import zipfile
+from typing import List
+
+from loguru import logger
+
+
+def zip_files(files: List[str], output_path: str) -> None:
+    with zipfile.ZipFile(output_path, "w", zipfile.ZIP_DEFLATED) as zipper:
+        for file in files:
+            logger.debug(f"Zipping {file}...")
+            zipper.write(file)
+
+    for file in files:
+        try:
+            os.remove(file)
+            logger.debug(f"Deleted file: {file}")
+        except OSError as e:
+            logger.debug(f"Error deleting file: {file} - {e}")
```

Comment:
As I said in the other PR, we should check if the file is opened (by some other service) before removing it, otherwise, we would lose all the next log lines until the service creates another log file.

Reply:
Any suggestion of how to do it ?
I did try the following approach but was unable to make it work:
![image](https://user-images.githubusercontent.com/1215497/221637013-d4a7519c-2b94-4876-ac64-fa84e0b1ecd8.png)

Reply:
Yeah, I'm not sure why your code is not working, makes sense to me. My only concern is if using `attrs` parameter to create the `info` dict is populating correctly the `open_files`. Have you tried calling `Process.open_files()` instead?

As an alternative, from the terminal it's usually done with `lsof <filename>`, maybe you could just call it from Python and check its return?

Reply:
Can we use delete_everything here instead of os.remove?

Reply:
@joaoantoniocardoso check the latest version